runtime/tchannel_client.go

```diff
@@ -53,12 +55,13 @@ type tchannelClient struct {
 	timeout           time.Duration
 	timeoutPerAttempt time.Duration
 	routingKey        *string
-	loggers           map[string]*zap.Logger
+	Loggers           map[string]*zap.Logger
+	Scopes            map[string]tally.Scope
```

Comment:
Yes, I'm debating which approach to take.  Thought I'd put up this diff for feedback.

Reply:
@clu One problem with having TChannelHandler interface that exposes Logger and Scope methods is that it will need to expose the endpoint.EndpointID, endpoint.HandlerID and endpoint.Method too.  Similarly for TChannelClient too, it would need add ClientID, etc to the interface too.  Using struct makes it easier to expose these fields and the Logger and Scope field too.

Reply:
why `EndpointID`, `HandlerID` and `Method` are exposed on the client? `ClientID` seems reasonable, what else need to be exposed for the client?

Reply:
To clarify:

The `TChannelEndpoint` would expose the `TChannelHandler`, `Logger()`, `Scope()`, `EndpointID`, `HandlerID` and `Method`.

The `TChannelClient` would expose the `TChannelCaller`, `Logger(method)`, `Scope(method)` and `ClientID`.